src/app/runtime_genesis_ledger/runtime_genesis_ledger.ml

```diff
@@ -160,18 +166,21 @@ let main accounts_json_file dir n constants_file =
   let ledger_path = genesis_dir ^/ "ledger" in
   let proof_path = genesis_dir ^/ "genesis_proof" in
   let constants_path = genesis_dir ^/ "genesis_constants.json" in
-  let%bind accounts = get_accounts accounts_json_file n in
+  let%bind accounts = get_accounts accounts_json_file num_accounts in
   let%bind () =
     match
       Or_error.try_with_join (fun () ->
           let open Or_error.Let_syntax in
-          let%map accounts = accounts in
+          let%map accounts, num_accounts = accounts in
           let ledger = generate_ledger ~directory_name:ledger_path accounts in
-          ledger )
+          (ledger, num_accounts) )
     with
-    | Ok ledger ->
+    | Ok (ledger, num_accounts) ->
         let genesis_constants =
           read_write_constants constants_file constants_path
+            ~f:(fun (genesis_constants : Genesis_constants.t) ->
+              (* Store the true number of accounts in the configuration. *)
+              {genesis_constants with fake_accounts_target= num_accounts} )
```

Comment:
Looks like `num_accounts` here includes real accounts as well but it needs to be the number of fake accounts only

Reply:
I was guided by previous uses of `fake_accounts_target` as total number of accounts, see `src/lib/genesis_ledger/testnet_postake_ledger.ml`. I'll change the name.

Reply:
Done.